build-tools/timing/timing.targets

```diff
@@ -63,15 +63,42 @@
     </PropertyGroup>
     <!--Simulate a clean checkout with git commands-->
     <Git
-        Arguments="rm -r --cached %(XACaptureBuildTimingProject.DirectoryToClean)"
+        Condition=" '%(XACaptureBuildTimingProject.Repo)' == '' "
+        Arguments="rm -r --cached &quot;%(XACaptureBuildTimingProject.DirectoryFullPath)&quot;"
```

Comment:
If `%(Repo)` is blank, the project is in the current repository, and doesn't need to be cloned.

These two commands are basically deleting the directory and recreating it:
```
git rm -r --cached samples/HelloWorld
git checkout HEAD samples/HelloWorld
```

Let's say I have staged changes for `samples/HelloWorld/MainActivity.cs`, and I didn't use `--cached`, I get the error:
```
error: the following file has changes staged in the index:
    samples/HelloWorld/MainActivity.cs
(use --cached to keep the file, or -f to force removal)
```
I guess I chose `--cached` to get past this error instead of `-f`, but the second `checkout` command is going to reset the changed file anyway...

That said, maybe I should just call the `<RemoveDir />` task instead?

Reply:
I think we should instead return to my [earlier comment](https://github.com/xamarin/xamarin-android/pull/1710#discussion_r189756390):

> It's a matter of expectations. If I have changes in a submodule, and running a make target/MSBuild target *nukes those changes*, I'm not a happy camper.

Similarly, if I have changes *in my local checkout* (e.g. more `Console.WriteLine()`s in `samples/HelloWorld`), and running some set of unit tests *nukes those changes*...  I will not be a happy camper.

*What is the **intent** here*?

We want build timing information on CI builds. In CI builds, there won't be any changes to worry about. If we want "fresh" build times, e.g. we're re-building a previously built project, we can `git clean -f samples/HelloWorld`.

`git clean -f` does *not* revert changes made to that directory.

Meanwhile, if we're on our own machines, *not* CI builds, and as part of some investigation we make local changes to see what happens, *we don't want those changes nuked*. `git clean -f` doesn't remove our changes (unless we add a file without `git add`). It is thus *safer* than `git rm -r ...; git checkout HEAD ...`. (It's *not* "safe," just "safer." It's still *much better* than using `git rm`.)

Reply:
I'm always making local `Console.WriteLine` changes. Having something auto nuke that will be a pain.

Reply:
`git clean` is working fine, so I changed it to this.

Also one of the projects we are testing is the `Xamarin.Forms.Performance.Integration` project, which is in `Xamarin.Android-Tests.sln`. We have to do something to get that one to a clean state, even on CI.